sql_generators/mobile_kpi_support_metrics/__init__.py

```diff
@@ -27,70 +28,118 @@
     "engagement.view.sql",
 )
 
-INSTALL_SOURCE = [
-    {
-        "name": "install_source",
-        "type": "STRING",
-        "description": "The source of a profile installation.",
-    },
-]
-
-ADJUST_FIELDS = [
-    {
-        "name": "adjust_ad_group",
-        "type": "STRING",
-        "description": "Adjust Ad Group the profile is attributed to.",
-    },
-    {
-        "name": "adjust_campaign",
-        "type": "STRING",
-        "description": "Adjust Campaign the profile is attributed to.",
-    },
-    {
-        "name": "adjust_creative",
-        "type": "STRING",
-        "description": "Adjust Creative the profile is attributed to.",
-    },
-    {
-        "name": "adjust_network",
-        "type": "STRING",
-        "description": "Adjust Network the profile is attributed to.",
-    },
-]
-
-FENIX_ATTRIBUTION_FIELDS = [
-    {
-        "name": "play_store_attribution_campaign",
-        "type": "STRING",
-        "description": "Play store campaign the profile is attributed to.",
-    },
-    {
-        "name": "play_store_attribution_medium",
-        "type": "STRING",
-        "description": "Play store Medium the profile is attributed to.",
-    },
-    {
-        "name": "play_store_attribution_source",
-        "type": "STRING",
-        "description": "Play store source the profile is attributed to.",
-    },
-    {
-        "name": "meta_attribution_app",
-        "type": "STRING",
-        "description": "Facebook app linked to paid marketing.",
-    },
-    *INSTALL_SOURCE,
-    *ADJUST_FIELDS,
-]
-
-FIREFOX_IOS_ATTRIBUTION_FIELDS = [
-    {
-        "name": "is_suspicious_device_client",
-        "type": "BOOLEAN",
-        "description": "Flag to identify suspicious device users, see bug-1846554 for more info.",
-    },
-    *ADJUST_FIELDS,
-]
+
+class Pings(Enum):
+    """TODO."""
+
+    metrics: str = "metrics"
+    first_session: str = "first_session"
+    baseline: str = "baseline"
+
+
+@dataclass
+class AttributionFieldGroup:
+    """TODO."""
+
+    name: str
+    source_pings: list[Pings]
+    fields: list[dict[str, str]]
+
+
+class AttributionFields:
```

Comment:
Interesting approach -- I'd still evaluate the option to move this fields to a JSON or YAML file for easier maintenance, because changes or additions wouldn't require code modification to the code. It's ok if it's in a later PR, though hopefully within the scope of the proposal.

Reply:
This would be a follow-up change and would be out of the scope of this PR. But yes, we should consider moving some of this stuff into an external configuration.